include/seqan3/alignment/matrix/alignment_coordinate.hpp

```diff
@@ -12,20 +12,346 @@
+ * \tparam states   A non-type template flag to enable a specific advanceable policy.
+ * \see seqan3::detail::advanceable_alignment_coordinate_state.
+ *
+ * \details
+ *
+ * This class provides all members to make the `advanceable_alignment_coordinate` be usable in a std::ranges::iota_view.
+ * For the purpose of alignments, modelling only incrementable and decrementable would fully suffice.
+ * Unfortunately, the current range implementation does not preserve std::ranges::BidirectionalRange properties,
+ * so we need to model the full advanceable concept in order to preserve the std::ranges::RandomAccessRange properties.
+ * This, however, can be relaxed if the range implementation fully complies with the current standard draft for Ranges,
+ * as increment and decrement would be enough and ranges::view::iota would preserve std::ranges::BidirectionalRange.
  */
-struct alignment_coordinate
+template <advanceable_alignment_coordinate_state state = advanceable_alignment_coordinate_state::none>
+class advanceable_alignment_coordinate
 {
-    //!\brief The position in the first sequence.
-    size_t seq1_pos;
-    //!\brief The position in the second sequence.
-    size_t seq2_pos;
+public:
+
+    //!\name Member types
+    //!\{
+    //!\brief Defines the difference type to model the std::WeaklyIncrementable concept.
+    using difference_type = std::make_signed_t<size_t>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr advanceable_alignment_coordinate() noexcept = default;
+    constexpr advanceable_alignment_coordinate(advanceable_alignment_coordinate const &) noexcept = default;
+    constexpr advanceable_alignment_coordinate(advanceable_alignment_coordinate &&) noexcept = default;
+    constexpr advanceable_alignment_coordinate & operator=(advanceable_alignment_coordinate const &) noexcept = default;
+    constexpr advanceable_alignment_coordinate & operator=(advanceable_alignment_coordinate &&) noexcept = default;
+    ~advanceable_alignment_coordinate() noexcept = default;
+
+    //!\brief Copy-constructs from another advanceable_alignment_coordinate with a different policy.
+    template <advanceable_alignment_coordinate_state _state>
+    //!\cond
+        requires !std::Same<_state, state>
+    //!\endcond
+    constexpr advanceable_alignment_coordinate(advanceable_alignment_coordinate<_state> const & other) :
+        first_seq_pos{other.first_seq_pos},
+        second_seq_pos{other.second_seq_pos}
+    {}
+
+    //!\brief Move-constructs from another advanceable_alignment_coordinate with a different policy.
+    template <advanceable_alignment_coordinate_state _state>
+    //!\cond
+        requires !std::Same<_state, state>
+    //!\endcond
+    constexpr advanceable_alignment_coordinate(advanceable_alignment_coordinate<_state> && other) :
+        first_seq_pos{std::move(other.first_seq_pos)},
+        second_seq_pos{std::move(other.second_seq_pos)}
+    {}
```

Comment:
So it is at the moment. Maybe it will change someday. if not moving it, it could have side effects if the position types change to something that is not builtin anymore. So from this perspective this is a very valid implementation of the move constructor.